Library/MapleBacon/MapleBacon/Resizer.swift

```diff
@@ -6,176 +6,251 @@ import UIKit
-        var imageToReturn = image
-        if let resizedImage = imageFromImage(image, toSize: size, interpolationQuality: quality),
-        let croppedImage = croppedImageFromImage(resizedImage, toBounds: bounds) {
+    
+    /**
+     Prequisite reszing method
+     */
+    private func resize(toSize size: CGSize, toBounds bounds: CGRect, interpolationQuality quality: CGInterpolationQuality, async: Bool, completion: ResizerCompletion) {
+        
+        var imageToReturn = self.image
+        
+        let drawTransposed: Bool
+        switch image.imageOrientation {
+        case .left, .leftMirrored, .right, .rightMirrored:
+            drawTransposed = true
+        default:
+            drawTransposed = false
+        }
+        
+        if
+            let resizedImage = resize(toSize: size, usingTransform: transformForOrientation(toSize: size), drawTransposed: drawTransposed, interpolationQuality: quality),
+            let croppedImage = crop(image: resizedImage, toBounds: bounds)
+        {
             imageToReturn = croppedImage
         }
-        if async {
-            dispatch_async(dispatch_get_main_queue()) {
-                completion(imageToReturn)
-            }
-        } else {
+      
+        let action = {
             completion(imageToReturn)
         }
+        
+        if async {
+            DispatchQueue.main.async(execute: action)
+            return
+        }
+        
+        action()
     }
 
-    private class func offsetFromSize(size: CGSize, toSize: CGSize) -> CGSize {
-        return CGSize(width: size.width - toSize.width, height: size.height - toSize.height)
-    }
-
-    private class func imageFromImage(image: UIImage, toSize size: CGSize,
-                                      interpolationQuality quality: CGInterpolationQuality) -> UIImage? {
-        let drawTransposed: Bool
-        switch (image.imageOrientation) {
-        case .Left, .LeftMirrored, .Right, .RightMirrored:
-            drawTransposed = true
-        default:
-            drawTransposed = false
+    /**
+     Actual resizing method
+     */
+    private func resize(toSize size: CGSize, usingTransform transform: CGAffineTransform, drawTransposed transpose: Bool, interpolationQuality quality: CGInterpolationQuality) -> UIImage? {
+        
+        let newRect = size.zeroBoundedRect().integral
+        let transposedRect = newRect.zeroBoundedRect()
+        
+        guard let imageRef: CGImage = image.cgImage else {
+            return nil
         }
-
-        return imageFromImage(image, toSize: size, usingTransform: transformForOrientationImage(image, toSize: size),
-                drawTransposed: drawTransposed, interpolationQuality: quality)
+        
+        let bpc = imageRef.bitsPerComponent
+        let bpr = imageRef.bytesPerRow * Int(UIScreen.main().scale)
+        let crs = imageRef.colorSpace!
+        let bmi = imageRef.bitmapInfo.rawValue
```

Comment:
the variables have a short scope but still I feel the names should rather spell out what they are